src/Microsoft.ML.Api/ApiUtils.cs

```diff
@@ -81,21 +97,58 @@ private static OpCode GetAssignmentOpCode(Type t)
             return mb.CreateDelegate(typeof(Peek<TRow, TValue>));
         }
 
+        private static Delegate GeneratePeek<TOwn, TRow, TValue>(PropertyInfo propertyInfo, OpCode assignmentOpCode)
+        {
+            // REVIEW: It seems like we really should cache these, instead of generating them per cursor.
+            Type[] args = { typeof(TOwn), typeof(TRow), typeof(long), typeof(TValue).MakeByRefType() };
+            var mb = new DynamicMethod("Peek", null, args, typeof(TOwn), true);
+            var il = mb.GetILGenerator();
+            var minfo = propertyInfo.GetGetMethod();
+            var opcode = minfo.IsVirtual ? OpCodes.Callvirt : OpCodes.Call;
```

Comment:
Do we need to handle `IsAbstract`?

Reply:
I'm not 100% sure, I thought IsAbstract implies IsVirtual, but the MSDN docs don't say that so I'll add it